### PR TITLE
[`flake8-simplify`]: combine-if-conditions

### PR DESCRIPTION
**Pull request by colin99d**
Ref: #998 

Still very much a WIP. @charliermarsh, I am thinking I will not add an autofixer on this one, because there are a lot of different ways the if statements could be formatted. I think it would be hard to do well. If you disagree, let me know.

### CODE REVIEW

**Comment by charliermarsh** [comment]
@colin99d - All good, no objections on my end.

**Comment by colin99d** [comment]
Screenshot 2023-02-10 at 6 16 38 PM
FYI. Our pre-commit linters are currently fighting over how to format the markdown. Let me know if I should open an issue.

File: crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs [hunk 1]
```diff
@@ -455,6 +485,84 @@ fn compare_expr(expr1: &ComparableExpr, expr2: &ComparableExpr) -> bool {
...
+    true
+}
+
+/// SIM114
+pub fn combine_if_conditions(checker: &mut Checker, body: &[Stmt], orelse: &[Stmt]) {
+    if orelse.is_empty() {
+        return;
+    }
+    // It's not all combinations because of this:
+    // https://github.com/MartinThoma/flake8-simplify/issues/70#issuecomment-924074984
```
Comment (spaceone) [inline crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs:545]: maybe write the reason directly here, if the link becomes unavailable somewhen

File: crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs [hunk 2]
```diff
@@ -92,6 +95,33 @@ impl Violation for UseTernaryOperator {
...
+    ///
+    /// ### Example
+    /// ```if x = 1:
+    ///     print("Hello")
+    /// elif x = 2:
+    ///     print("Hello")
+    /// ```
+    ///
+    /// Use instead:
+    /// ```if x = 1 or x = 2
```
Comment (spaceone) [inline crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs:113]:
```suggestion
    /// ```python
    /// if x = 1:
    ///     print("Hello")
    /// elif x = 2:
    ///     print("Hello")
    /// ```
    ///
    /// Use instead:
    /// ```python
    /// if x = 1 or x = 2
```

**Comment by charliermarsh** [comment]
For some reason I'm having trouble pushing edits... Maybe related to the HEAD being deleted and the PR closed then re-opened. Do you mind closing this and reopening as a completely new PR?

**Comment by charliermarsh** [comment]
I'm gonna apply some edits in a follow-up PR, just easier.

**Pull request closed without merging**